File: week1/3.decimal_to_binary/go/decimal_to_binary.go
package main

import (
	"fmt"
)

func main() {
	res := binaryToDecimal(42)
	fmt.Println(res)
}

func binaryToDecimal (n int) int {
	 var bin int = 0
	 i := 1

	 for n > 0 {
		bin = bin + (n % 2) * i;
		n = (n / 2)
		i *= 10
	 }

	 return bin
}

// OUTPUT- 101010

File: week1/3.decimal_to_binary/dart/decimal_to_binary.dart
void main() {
  int res = binaryToDecimal(10);
  print(res);
}

int binaryToDecimal(int n) {
  int bin = 0;
  int i = 1;

  while (n > 0) {
    bin = bin + (n % 2) * i;
    n = (n / 2).floor();
    i *= 10;
  }

  return bin;
}



